### PR TITLE
pids: re-enabled tests on Alpine Linux

### LINKED ISSUE #4268: pids: tests do not work on Alpine linux
### Summary
Currently the tests for pids do not work on Alpine linux, which is supported in CI since #4222.

### Issue Type

Bug Report

### Component Name

pids

### Ansible Version

unrelated

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Fixing pids test on Alpine Linux

Fixes #4268

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
tests/integration/targets/pids/tasks/main.yml

### CODE REVIEW

**Comment by russoz** [comment]
The test relies on copying the `sleep` binary in the system to a temporary location and running it with a different name. On Alpine Linux, however, `sleep` is actually a link to `busybox` and its behaviour is defined by the name it is called, hence the results are totally different.

Will try and replace that logic with something else that does not depend on copying the `sleep` binary.

**Comment by russoz** [comment]
Fixes #4268

File: tests/integration/targets/pids/tasks/main.yml [hunk 1]
```diff
@@ -7,112 +7,105 @@
...
+    dest: "{{ remote_tmp_dir }}/obtainpid.sh"
+    mode: 0755
+
+- name: "Run the fake 'sleep' binary"
+  command: "sh {{ remote_tmp_dir }}/obtainpid.sh '{{ remote_tmp_dir }}/{{ random_name }}' '{{ remote_tmp_dir }}/obtainpid.txt'"
+
+  async: 100
+  poll: 0
+
+- name: "Wait for one second to make sure that the sleep copy has actually been started"
```
Comment (felixfontein) [inline tests/integration/targets/pids/tasks/main.yml:57]:
```suggestion
- name: "Wait for one second to make sure that the fake 'sleep' binary has actually been started"
```

File: tests/integration/targets/pids/tasks/main.yml [hunk 2]
```diff
@@ -7,112 +7,105 @@
...
+  pause:
+    seconds: 1
+
+- name: "Checking the process IDs (PIDs) of sleep binary"
```
Comment (felixfontein) [inline tests/integration/targets/pids/tasks/main.yml:61]:
```suggestion
- name: "Checking the process IDs (PIDs) of fake 'sleep' binary"
```

**Review by felixfontein — COMMENTED** [review]
For consistency :)

**Comment by russoz** [comment]
@felixfontein it just occurred to me: should I add that license preamble to the .c file as well? If nothing else, for consistency ;-)

**Comment by felixfontein** [comment]
@russoz that's probably a good thing!

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for fixing this!

**Pull request merged**